optaplanner-core/src/main/java/org/optaplanner/core/api/score/holder/AbstractScoreHolder.java

```diff
@@ -110,7 +124,7 @@ private ConstraintMatchTotal findConstraintMatchTotal(RuleContext kcontext) {
         String constraintName = rule.getName();
         String constraintId = constraintPackage + "/" + constraintName;
         return constraintMatchTotalMap.computeIfAbsent(constraintId,
-                k -> new ConstraintMatchTotal(constraintPackage, constraintName, zeroScore));
+                k -> new ConstraintMatchTotal(constraintPackage, constraintName, null, zeroScore));
```

Comment:
Let's use the constructor taking only 3 parameters to minimize passing null values.

Reply:
I want to explicitly use the null parameter here so it sticks out like a sore thumb, because I hope it's possible to get rid of it some day.